jumpcloud_bootstrap_template.sh

```diff
@@ -1196,21 +1598,153 @@ if [[ -f $DEP_N_GATE_DONE ]]; then
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): The following errors were returned when attempting to delete the users:" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): ################################  Error Details ################################" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): $commandResponseContent" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): ################################################################################" >>"$DEP_N_DEBUG"
         fi
+        errorNF='not found'
+        # errorNF='User \w+ not found'
+        if [[ $commandResponseContent =~ $errorNF ]]; then
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): [ERROR] sysadminctl encountered a user not found error, the $user1 && $user2 users may not be deleted from the system" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): The following errors were returned when attempting to delete the users:" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): ################################  Error Details ################################" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): $commandResponseContent" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): ################################################################################" >>"$DEP_N_DEBUG"
+        fi
+        ## end parse results
 
+        ## delete Command:
+        cmdCreateRequest=$(
+            curl -s -o $DEP_N_HTTP_RESPONSE -w "%{http_code}" \
+            -X DELETE \
+            -H 'Accept: application/json' \
+            -H 'Content-Type: application/json' \
+            -H "x-api-key: $APIKEY" \
+                --url https://console.jumpcloud.com/api/commands/$commandID
+        )
+        responseContent=$(cat $DEP_N_HTTP_RESPONSE)
+        if [[ $cmdCreateRequest != "200" ]]; then
+            # handle error
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): [error] HTTP Response does not indicate success" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): HTTP Response: $responseContent" >>"$DEP_N_DEBUG"
+        else
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): Command used to trigger the user cleanup was removed" >>"$DEP_N_DEBUG"
+        fi
+        # Wait 10 seconds for user tables to update
+        echo "$(date "+%Y-%m-%d %H:%M:%S"): Wait 30s" >>"$DEP_N_DEBUG"
+        sleep 30
+        function validateUser() {
+            userStatus=$([[ -n $(id -u "$1" 2>/dev/null) ]] && echo 1 || echo 0);
+            if [[ $userStatus == 0 ]];then
+                echo "$(date "+%Y-%m-%d %H:%M:%S"): User: $1 has been removed from system" >>"$DEP_N_DEBUG"
+            else
+                echo "$(date "+%Y-%m-%d %H:%M:%S"): [ERROR] User: $1 was not removed from system" >>"$DEP_N_DEBUG"
+            fi
+        }
+        # Validate that the users were removed from the system
+        validateUser "$ENROLLMENT_USER"
```

Comment:
Here both enrollment users, the first user and the decryption user should be noted in the logs 
![Screen Shot 2023-07-26 at 2 11 29 PM](https://github.com/TheJumpCloud/MDM-Prestage-User-Enrollment/assets/54448601/5b0604e6-4bc3-4ad5-8b55-ae213427fe4d)